src/cmd/ksh93/tests/builtins.sh

```diff
@@ -476,7 +476,15 @@ $SHELL <<- \EOF
     cd
     [[ $(pwd) == "$home" ]]
 EOF
-[[ $? == 0 ]] || log_error 'cd with no arguments fails if HOME is unset'
+if [[ $? != 0 ]]
+then
+    if [[ ! -d $home ]]
```

Comment:
This is wrong. The `home` var isn't visible in this scope since it was defined in a subshell. Also, we generally want to skip a test if we know it is going to fail. Rather than deal with a failure by deciding if the test should have been skipped. I can't think of any other tests where we use this pattern.

I wonder if the real issue here is that `$USER` is undefined in your test environment. It's oftentimes useful to add a `set -x` at the top of the test script.

Reply:
Yes, you're right, home shouldn't be visible here, let me rework this.

Reply:
I would like to see the output of `log_info "HOME=|$HOME| USER=|$USER|"` immediately prior to the failing test

Reply:
<I> builtins[517]: HOME=|/tmp/ksh.builtins.Eflf2W/home| USER=|anuradha|